### PR TITLE
Nix/rnix update

### PR DESCRIPTION
**Pull request by francois-caddet**
To try to fix some strange bugs and because the API is a lot cleaner, this branch update `rnix` from 0.10 to 0.11. with a lot of breakage in the API but with a good improvement.

### CODE REVIEW

File: src/nix.rs [hunk 1]
```diff
@@ -250,53 +238,49 @@ impl ToNickel for rnix::SyntaxNode {
...
-            // The following nodes aren't `RichTerm`s but sub-parts of other nodes higher in the syntax tree.
-            // They can't be parsed independently of a term and should never occur at this level of the translation.
-            ParsedType::Pattern(_)
-            | ParsedType::PatEntry(_)
-            | ParsedType::Inherit(..)
-            | ParsedType::InheritFrom(..)
-            | ParsedType::Key(..)
-            | ParsedType::KeyValue(..) => unreachable!(),
+            Expr::Path(_) | Expr::HasAttr(_) => unimplemented!(), // TODO: What is the diff with
+                                                                  // Select / operator HasField?
```
Comment (yannham) [inline src/nix.rs:275]: I didn't understand this comment.

File: src/nix.rs [hunk 2]
```diff
@@ -250,53 +238,49 @@ impl ToNickel for rnix::SyntaxNode {
...
         .with_pos(crate::position::TermPos::Original(span))
     }
 }
 
 pub fn parse(cache: &Cache, file_id: FileId) -> Result<RichTerm, rnix::parser::ParseError> {
     let source = cache.files().source(file_id);
-    let nixast = rnix::parse(source).as_result()?;
-    Ok(nixast.node().to_nickel(file_id))
+    let root = rnix::Root::parse(source).ok()?; // TODO: we could return a list of errors calling
+                                                // `errors()` could improve error management.
```
Comment (yannham) [inline src/nix.rs:284]:
```suggestion
    let root = rnix::Root::parse(source).ok()?; // TODO: we could return a list of errors calling
                                                // `errors()` to improve error management.
```

File: src/nix.rs [hunk 3]
```diff
@@ -60,80 +69,59 @@ impl ToNickel for BinOp {
...
-        let node: ParsedType = self.try_into().unwrap();
-        match node {
-            ParsedType::Error(_) => {
+        match self {
+            Expr::Error(_) => {
                 Term::ParseError(crate::error::ParseError::NixParseError(file_id)).into()
+                // TODO:
+                // Improve
+                // error
+                // management
```
Comment (yannham) [inline src/nix.rs:86]:
```suggestion
                // TODO: Improve error management
```

**Review by yannham — APPROVED** [review]
Looks good overall. As long as the tests pass, you can merge this now and worry about clippy on the nix/nickel branch directly.

File: src/nix.rs (hunk 1 above)
Comment (francois-caddet) [reply src/nix.rs:275]: Sorry:

- For the `HasAttr` I discovered later that it's not an operator anymore in `rnix 0.11`. It was an operator before but now it's a specific language construct.
- But for `Expr::Path` I don't understand what it is for. we already have `Select` which is the `.` operator. So what is a `Expr::Path`? 

`rnix` is a very good crate but sadly without any documentation...

Comment (yannham) [reply src/nix.rs:272]: @francois-caddet I suppose it's a file path, like `/foo/bar/baz`, not something related to attribute paths?

Comment (francois-caddet) [reply src/nix.rs:272]: Yes but so what is `Literal::Uri(_)`? 😢 
I will merge as it is and see what it is adding tests in the main branch `nix/nickel`

**Pull request merged**